core/src/main/java/io/questdb/griffin/model/RuntimeIntervalModelBuilder.java

```diff
@@ -125,8 +125,20 @@ public void setBetweenNegated(boolean isNegated) {
 
     public void setBetweenBoundary(long timestamp) {
         if (!betweenBoundarySet) {
-            betweenBoundary = timestamp;
-            betweenBoundarySet = true;
+            if (timestamp == Numbers.LONG_NaN) {
+                if (!betweenNegated) {
+                    intersectEmpty();
+                }
+                // else {
+                // NOT BETWEEN with NULL
+                // to be consistent with non-designated filtering
+                // do no filtering
+                // }
+                betweenBoundarySet = false;
```

Comment:
I think it should be `betweenBoundarySet = true;`

Try `WHERE timestamp between NULL and '2020-01-01'`

Reply:
Actually remove this if: `if (timestamp == Numbers.LONG_NaN) {` and leave else as it was. Below code handles Numbers.LONG_NaN boundaries as well as if second boundary is dynamic